### PR TITLE
Feature/news update

### PR DESCRIPTION
**Pull request by Delf-Lee**
### 이런게 변경되었어요. 💍
- 뉴스를 수정할 수 있게 되었습니다.

### CODE REVIEW

File: src/main/java/com/snack/news/controller/AdminController.java [hunk 1]
```diff
@@ -7,9 +7,11 @@
 import lombok.AllArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
 import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
+import java.nio.file.Watchable;
```
Comment (NESOY) [inline src/main/java/com/snack/news/controller/AdminController.java]: Codacy Issue found: [Avoid unused imports such as 'java.nio.file.Watchable'](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

**Comment by NESOY** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Snack-BE/Snack-BE/pullRequest?prid=4554642&bid=15319529)

File: src/main/java/com/snack/news/controller/AdminController.java [hunk 2]
```diff
@@ -7,9 +7,11 @@
 import lombok.AllArgsConstructor;
 import org.springframework.data.domain.Page;
 import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
```
Comment (NESOY) [inline src/main/java/com/snack/news/controller/AdminController.java]: Codacy Issue found: [Unused import - org.springframework.transaction.annotation.Transactional.](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

File: src/test/java/com/snack/news/service/AdminServiceTest.java [hunk 3]
```diff
@@ -17,9 +17,9 @@
 
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 
-import static org.mockito.ArgumentMatchers.any;
-import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.*;
```
Comment (NESOY) [inline src/test/java/com/snack/news/service/AdminServiceTest.java]: Codacy Issue found: [Avoid unused imports such as 'org.mockito.ArgumentMatchers'](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

File: src/main/java/com/snack/news/controller/AdminController.java (hunk 2 above)
Comment (NESOY) [inline src/main/java/com/snack/news/controller/AdminController.java]: Codacy Issue found: [Avoid unused imports such as 'org.springframework.transaction.annotation.Transactional'](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

File: src/main/java/com/snack/news/controller/AdminController.java (hunk 1 above)
Comment (NESOY) [inline src/main/java/com/snack/news/controller/AdminController.java]: Codacy Issue found: [Unused import - java.nio.file.Watchable.](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

File: src/main/java/com/snack/news/domain/news/News.java [hunk 4]
```diff
@@ -4,10 +4,8 @@
 import com.snack.news.domain.category.Category;
 import com.snack.news.domain.tag.Tag;
 import com.snack.news.domain.topic.Topic;
-import lombok.Builder;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.ToString;
+import com.snack.news.dto.NewsDto;
+import lombok.*;
```
Comment (NESOY) [inline src/main/java/com/snack/news/domain/news/News.java]: Codacy Issue found: [Avoid unused imports such as 'lombok'](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

File: src/main/java/com/snack/news/domain/news/News.java [hunk 5]
```diff
@@ -4,10 +4,8 @@
 import com.snack.news.domain.category.Category;
 import com.snack.news.domain.tag.Tag;
 import com.snack.news.domain.topic.Topic;
-import lombok.Builder;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
-import lombok.ToString;
+import com.snack.news.dto.NewsDto;
```
Comment (NESOY) [inline src/main/java/com/snack/news/domain/news/News.java]: Codacy Issue found: [Unused import - com.snack.news.dto.NewsDto.](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

Comment (NESOY) [inline src/main/java/com/snack/news/domain/news/News.java]: Codacy Issue found: [Avoid unused imports such as 'com.snack.news.dto.NewsDto'](https://app.codacy.com/app/snack-news/Snack-BE/pullRequest?prid=4554642)

**Comment by chul-hyun** [comment]
Hey @Delf-Lee,  
Your changes look good to me!

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;snack-news&#x2F;Snack-BE&#x2F;builds&#x2F;616152249">View build log</a>

###### TravisBuddy Request Identifier: 9d2f0d90-0e5f-11ea-aab6-13cb450ce62c

File: src/main/java/com/snack/news/controller/AdminController.java [hunk 6]
```diff
@@ -25,21 +27,26 @@
 
 	@GetMapping("/news")
 	public ResponseEntity<Page<News>> getNewsList() {
-		return getNewsList(1);
+		return getNewsList(1L);
```
Comment (NESOY) [inline src/main/java/com/snack/news/controller/AdminController.java]:
```suggestion
		final long DEFAULT_PAGE_SIZE = 1L;
		return getNewsList(DEFAULT_PAGE_SIZE);
```

File: src/main/java/com/snack/news/controller/AdminController.java [hunk 7]
```diff
@@ -25,21 +27,26 @@
...
 	}
 
 	@GetMapping("/news/{page}")
-	public ResponseEntity<Page<News>> getNewsList(@PathVariable int page) {
+	public ResponseEntity<Page<News>> getNewsList(@PathVariable long page) {
```
Comment (NESOY) [inline src/main/java/com/snack/news/controller/AdminController.java:33]: 👍

File: src/main/java/com/snack/news/domain/news/News.java [hunk 8]
```diff
@@ -51,13 +49,25 @@
 	private LocalDateTime publishAt;
 
 	@Builder
-	public News(String title, String link, String content, Category category, List<Topic> topics, List<Tag> tags, LocalDateTime publishAt) {
+	public News(String title, String content, String link, LocalDateTime publishAt, Category category, List<Topic> topics, List<Tag> tags) {
```
Comment (NESOY) [inline src/main/java/com/snack/news/domain/news/News.java:54]: 순서를 바꾸신 이유가 있으실까요?

File: src/main/java/com/snack/news/service/AdminService.java [hunk 9]
```diff
@@ -29,21 +29,27 @@
...
+		News news = generateNews(newsDto);
 		newsRepository.save(news);
 
 		return NewsDto.builder().id(news.getId()).build();
 	}
 
-	public Page<News> getNewsList(int page) {
-		Pageable pageable = PageRequest.of(page - 1, 10, SORT_BY_ID);
+	public Page<News> getNewsList(long page) {
+		Pageable pageable = PageRequest.of((int) page - 1, 10);
```
Comment (NESOY) [inline src/main/java/com/snack/news/service/AdminService.java]: `SORT_BY_ID`가 제거 되었는데 상수도 같이 지워주는건 어떠신가요?

**Comment by chul-hyun** [comment]
Hey @Delf-Lee,
Something went wrong with the build.

TravisCI finished with status `errored`, which means the build failed because of something unrelated to the tests, such as a problem with a dependency or the build process itself.

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;snack-news&#x2F;Snack-BE&#x2F;builds&#x2F;616155804">View build log</a>

###### TravisBuddy Request Identifier: fb312520-0e61-11ea-aab6-13cb450ce62c

**Comment by NESOY** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Snack-BE/Snack-BE/pullRequest?prid=4554642&bid=15319529)

**Comment by NESOY** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/Snack-BE/Snack-BE/pullRequest?prid=4554642&bid=15319529)

**Comment by chul-hyun** [comment]
Hey @Delf-Lee,  
Your changes look good to me!

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;snack-news&#x2F;Snack-BE&#x2F;builds&#x2F;616156523">View build log</a>

###### TravisBuddy Request Identifier: 7de672e0-0e62-11ea-aab6-13cb450ce62c

File: src/main/java/com/snack/news/service/AdminService.java [hunk 10]
```diff
@@ -52,4 +58,27 @@ public void deleteNews(long newsId) {
...
+		List<Topic> topics = topicService.getTopicList(newsDto.getTopicIds());
+		List<Tag> tags = tagService.getTagList(newsDto.getTagIds());
+
+		return newsDto.toEntity(category, topics, tags);
+	}
+
+	private News updateNews(News news, NewsDto newsDto) {
+		Category category = categoryService.getCategory(newsDto.getCategoryId());
+		List<Topic> topics = topicService.getTopicList(newsDto.getTopicIds());
+		List<Tag> tags = tagService.getTagList(newsDto.getTagIds());
```
Comment (NESOY) [inline src/main/java/com/snack/news/service/AdminService.java:74]: `generateNews`의 3줄과 비슷한 로직을 수행하는거 같네요!
News에 필요한 정보를 수집하는 책임으로 보이네요 ㅎㅎ 만약에 함수로 추출한다면 어떤 이름이 적당할까요?
아니면 JPA에서 객체를 가져올때 여러 테이블을 조인해서 정보를 한번에 가져올 수는 없을까요?
불필요한 부분을 Lazy로 안가져올 수 있지 않을까요?

**Comment by chul-hyun** [comment]
Hey @Delf-Lee,  
Your changes look good to me!

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;snack-news&#x2F;Snack-BE&#x2F;builds&#x2F;616157817">View build log</a>

###### TravisBuddy Request Identifier: 483119b0-0e63-11ea-aab6-13cb450ce62c

**Review by NESOY — APPROVED** [review]

File: src/main/java/com/snack/news/domain/news/News.java (hunk 8 above)
Comment (Delf-Lee) [reply src/main/java/com/snack/news/domain/news/News.java:54]: 주 목적은 단순히 "직관적인 배치" 였고, 표현한다면 News에 있어서 중요한 데이터를 앞으로, 메타데이터를 뒤로 옮긴 정도겠네요.

**Comment by chul-hyun** [comment]
Hey @Delf-Lee,  
Your changes look good to me!

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;snack-news&#x2F;Snack-BE&#x2F;builds&#x2F;616160405">View build log</a>

###### TravisBuddy Request Identifier: 66a31130-0e65-11ea-aab6-13cb450ce62c

File: src/main/java/com/snack/news/service/AdminService.java (hunk 10 above)
Comment (Delf-Lee) [reply src/main/java/com/snack/news/service/AdminService.java:74]: 해당 부분은 새 이슈로 남겨두었습니다.
https://github.com/snack-news/Snack-BE/issues/121#issue-527641967

**Comment by chul-hyun** [comment]
Hey @Delf-Lee,  
Your changes look good to me!

<a href="https:&#x2F;&#x2F;travis-ci.org&#x2F;snack-news&#x2F;Snack-BE&#x2F;builds&#x2F;616163874">View build log</a>

###### TravisBuddy Request Identifier: a772aec0-0e68-11ea-aab6-13cb450ce62c

**Pull request merged**